src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -1913,6 +1913,92 @@ public void TraceIdCustomGenerationTest()
             }).Dispose();
         }
 
+        [Fact]
+        public void EnumerateTagObjectsTest()
+        {
+            Activity a = new Activity("Root");
+
+            var enumerator = a.EnumerateTagObjects();
+
+            Assert.False(enumerator.MoveNext());
+            Assert.False(enumerator.GetEnumerator().MoveNext());
+
+            a.SetTag("key1", "value1");
+            a.SetTag("key2", "value2");
```

Comment:
Could you please add to these test the validation of the values returned from the properties against the values returned from the methods? I mean like Activity.TagObjects, Activity.Links, and Activity.Events?

Reply:
Good call, updated!